### PR TITLE
Set go_package in all proto files

### PR DESCRIPTION
**Pull request by thockin**
Protobuf docs recommend setting this.  When we do, we find that the hand-written code used the wrong package name.

This also creates some diff to the *.pb.go files to note that in the "options".

You can dump the gzipped blob with the following program (thanks StackOverflow!):

 ```go
package main

import (
        "bytes"
        "compress/gzip"
        "encoding/json"
        "fmt"
        "os"

        "io/ioutil"

        proto "github.com/golang/protobuf/proto"
        dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
)

func main() {
        m := map[string][]byte{
                "before": blobv1,
                "after": blobv2,
        }
        arg := os.Args[1]
        dump(m[arg])
}

func dump(bytes []byte) {
        fd, err := decodeFileDesc(bytes)
        if err != nil {
                panic(err)
        }
        b, err := json.MarshalIndent(fd, "", "  ")
        if err != nil {
                panic(err)
        }
        fmt.Println(string(b))
}

// decompress does gzip decompression.
func decompress(b []byte) ([]byte, error) {
        r, err := gzip.NewReader(bytes.NewReader(b))
        if err != nil {
                return nil, fmt.Errorf("bad gzipped descriptor: %v", err)
        }
        out, err := ioutil.ReadAll(r)
        if err != nil {
                return nil, fmt.Errorf("bad gzipped descriptor: %v", err)
        }
        return out, nil
}

func decodeFileDesc(enc []byte) (*dpb.FileDescriptorProto, error) {
        raw, err := decompress(enc)
        if err != nil {
                return nil, fmt.Errorf("failed to decompress enc: %v", err)
        }

        fd := new(dpb.FileDescriptorProto)
        if err := proto.Unmarshal(raw, fd); err != nil {
                return nil, fmt.Errorf("bad descriptor: %v", err)
        }
        return fd, nil
}
```

/kind cleanup
/sig node
/sig architecture

```release-note
NONE
```

### CODE REVIEW

**Comment by enj** [comment]
/sig auth

No clue what this is about 🙃

File: staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto [hunk 1]
```diff
@@ -2,6 +2,7 @@
 syntax = "proto3";
 
 package deviceplugin;
```
Comment (aojea) [inline staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto:4]: v1alpha1???

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1/api.proto [hunk 2]
```diff
@@ -2,6 +2,7 @@
 syntax = "proto3";
 
 package pluginregistration;
```
Comment (aojea) [inline staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1/api.proto:4]: v1?

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto [hunk 3]
```diff
@@ -2,6 +2,7 @@
 syntax = "proto3";
 
 package pluginregistration;
```
Comment (aojea) [inline staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto:4]: I still can't figure out what is the right value, it seems you have some places this matches the go module (version), and other places the versioned object

File: staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto (hunk 1 above)
Comment (thockin) [reply staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto:4]: yep, will fix

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto (hunk 3 above)
Comment (thockin) [reply staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto:4]: Convention seems to be the same as the Go package.  Other than the ones I missed, are there others?

Comment (aojea) [reply staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto]: I think I flagged all in the comments

**Comment by aojea** [comment]
the module name changes need to be propagated to other files too

**Comment by aojea** [comment]
/hold
 to stop retesting
@thockin the failure is legit

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.pb.go [hunk 4]
```diff
@@ -324,7 +326,7 @@ func NewRegistrationClient(cc *grpc.ClientConn) RegistrationClient {
 
 func (c *registrationClient) GetInfo(ctx context.Context, in *InfoRequest, opts ...grpc.CallOption) (*PluginInfo, error) {
 	out := new(PluginInfo)
-	err := c.cc.Invoke(ctx, "/pluginregistration.Registration/GetInfo", in, out, opts...)
+	err := c.cc.Invoke(ctx, "/v1beta1.Registration/GetInfo", in, out, opts...)
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.pb.go:329]: I don't know enough about grpc to know if this means the endpoint or wire format / content is changing... is it?

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1alpha1/api.pb.go [hunk 5]
```diff
@@ -324,7 +326,7 @@ func NewRegistrationClient(cc *grpc.ClientConn) RegistrationClient {
 
 func (c *registrationClient) GetInfo(ctx context.Context, in *InfoRequest, opts ...grpc.CallOption) (*PluginInfo, error) {
 	out := new(PluginInfo)
-	err := c.cc.Invoke(ctx, "/pluginregistration.Registration/GetInfo", in, out, opts...)
+	err := c.cc.Invoke(ctx, "/v1alpha1.Registration/GetInfo", in, out, opts...)
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1alpha1/api.pb.go:329]: same question about whether this means the endpoint or wire format/content is changing

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1/api.pb.go [hunk 6]
```diff
@@ -324,7 +325,7 @@ func NewRegistrationClient(cc *grpc.ClientConn) RegistrationClient {
 
 func (c *registrationClient) GetInfo(ctx context.Context, in *InfoRequest, opts ...grpc.CallOption) (*PluginInfo, error) {
 	out := new(PluginInfo)
-	err := c.cc.Invoke(ctx, "/pluginregistration.Registration/GetInfo", in, out, opts...)
+	err := c.cc.Invoke(ctx, "/v1.Registration/GetInfo", in, out, opts...)
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1/api.pb.go:328]: same question about whether this means the endpoint or wire format/content is changing

File: staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.pb.go [hunk 7]
```diff
@@ -607,7 +609,7 @@ func NewRegistrationClient(cc *grpc.ClientConn) RegistrationClient {
 
 func (c *registrationClient) Register(ctx context.Context, in *RegisterRequest, opts ...grpc.CallOption) (*Empty, error) {
 	out := new(Empty)
-	err := c.cc.Invoke(ctx, "/deviceplugin.Registration/Register", in, out, opts...)
+	err := c.cc.Invoke(ctx, "/v1alpha.Registration/Register", in, out, opts...)
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.pb.go:612]: same question about whether this means the endpoint or wire format/content is changing

**Review by liggitt — COMMENTED** [review]
changes that added or switched to the fully-qualified package and only changed the descriptor lgtm

the ones that changed the package name and modified the generated clients I'm less sure about

also, all the CSI tests are failing across both e2e jobs, which seems suspicious

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.pb.go (hunk 4 above)
Comment (thockin) [reply staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.pb.go:329]: yes, I'm not sure either.  I will revisit this PR.

**Comment by thockin** [comment]
I reverted the proto package name parts - Those are API breaks.

File: staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto [hunk 8]
```diff
@@ -1,7 +1,8 @@
 // To regenerate api.pb.go run `hack/update-codegen.sh protobindings`
 syntax = "proto3";
 
-package pluginregistration;
+package pluginregistration; // This should have been v1beta1.
```
[the remark this answers is not in the document]
Comment (thockin) [reply staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto:4]: The problem, I think, is that we used the same package name in different revisions of the proto API (v1alpha1, v1beta1, v1) - I don't know how a client would spec which to use, but I am not a grpc expert.

[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/kubelet/pkg/apis/pluginregistration/v1beta1/api.proto:4]: given the api.proto files are (coincidentally? luckily?) identical between pluginregistration v1alpha1 → v1beta1 → v1, maybe it doesn't matter?

**Pull request merged**